nettacker/database/db.py

```
@@ -290,9 +589,40 @@ def get_scan_result(id):
+                "SELECT report_path_filename from reports WHERE id = ?",
+                (id,),
+            )
 
-    return filename, open(str(filename), "rb").read()
+            row = cursor.fetchone()
+            if row:
+                filename = row[0]
+                try:
+                    return filename, open(str(filename), "rb").read()
+                except IOError as e:
+                    logger.error(f"Failed to read report file: {e}")
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)